test/TestApps/ToastNotificationsTestApp/main.cpp

```diff
@@ -1288,6 +1288,34 @@ bool VerifyRemoveAllAsync()
     return true;
 }
 
+
+bool VerifyForegroundRegistrationSuccess()
+{
+    winrt::AppNotificationManager::Default().Unregister();
```

Comment:
do you need a test if you call unregisterAll and then register notificationinvoked?

Reply:
The heart of the test is to register foreground handlers before you register. Unregister/UnregisterAll are interchangable.